aten/src/ATen/core/Half-inl.h

```diff
@@ -43,6 +43,14 @@ inline AT_HOSTDEVICE Half::operator __half() const {
 }
 #endif
 
+// CUDA intrinsics
+
+#if defined(__CUDA_ARCH__) && (__CUDA_ARCH__ >= 350)
+inline __device__ Half __ldg(const Half* ptr) {
+    return __ldg(reinterpret_cast<const __half*>(ptr));
```

Comment:
can we have the not have the declaration guarded by __CUDA_ARCH__ >= 350, but the definition? (you can just leave out the __ldg for other CUDA_ARCHs).  It's too easy to write code that doesn't compile this way.